### PR TITLE
Add codecov

### PR DESCRIPTION
**Pull request by manujz**
# Pull Request

## Related issue
#474 

## What does this PR do?
- Added codecov for codecoverage status on every pr.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .circleci/config.yml
Comment (brunoocasali) [inline .circleci/config.yml:1]: We're not using Circle CI you can remove this file :)

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -63,3 +63,15 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with: { go-version: "1.17" }
+      - run: |
+          go test -race -coverprofile=coverage.txt -covermode=atomic ./...
+      - uses: codecov/codecov-action@v3
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (brunoocasali) [inline .github/workflows/tests.yml:77]: I think you don't need a new job. Just add the coverage into the current `go test` call ;)

Comment (brunoocasali) [reply .github/workflows/tests.yml:77]: I'll add the token in the settings!

**Review by brunoocasali — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (manujz) [reply .github/workflows/tests.yml]: Done

**Review by brunoocasali — APPROVED** [review]
Thaanks! :D 

bors merge

**Comment by manujz** [comment]
@brunoocasali the bors check is stuck I guess?

**Pull request merged**